test/extensions/http/header_validators/envoy_default/path_normalizer_fuzz_test.cc

```diff
@@ -0,0 +1,73 @@
+#include "source/common/http/header_map_impl.h"
+#include "source/extensions/http/header_validators/envoy_default/http1_header_validator.h"
+
+#include "test/extensions/http/header_validators/envoy_default/path_normalizer_fuzz.pb.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/stream_info/mocks.h"
+
+namespace Envoy {
+
+// Fuzz the path normalization code.
+DEFINE_PROTO_FUZZER(
+    const test::extensions::http::header_validators::envoy_default::PathNormalizerFuzzTestCase&
+        input) {
+  auto header_map = Http::RequestHeaderMapImpl::create();
+  Http::HeaderString method;
+  Http::HeaderString path;
+  method.setCopyUnvalidatedForTestOnly(input.method());
+  path.setCopyUnvalidatedForTestOnly(input.path());
+
+  NiceMock<Envoy::StreamInfo::MockStreamInfo> stream_info;
+  Extensions::Http::HeaderValidators::EnvoyDefault::Http1HeaderValidator validator(
+      input.config(), Http::Protocol::Http11, stream_info);
+  // The character set of the :path and :method headers is validated before normalization.
+  // Here we will just not run the test with invalid values
+  if (!validator.validateMethodHeader(method) ||
+      (!(method == "CONNECT" && path.empty()) && !validator.validatePathHeaderCharacters(path))) {
```

Comment:
I think `(!(method == "CONNECT" && path.empty()) && !validator.validatePathHeaderCharacters(path)))`
should be 
`(!(method == "CONNECT" && path.empty()) || !validator.validatePathHeaderCharacters(path)))`

Since the validations seem orthogonal

Reply:
I only want skip calling `validatePathHeaderCharacters` in case method is `CONNECT` and path is empty. In this case the && should be correct. If I change it || the `validatePathHeaderCharacters`  will still be called with empty path for CONNECT method.

Reply:
sg, thanks for the explaination